test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -238,8 +249,22 @@ public void ParseUnquotedValue ()
             // TODO: is it possible to get the system to recognize when a complete unicode char is present and start the next one then, without a space?
 //            Assert.Equal("日本", Parsers.UnquotedValue.Parse(@"\xe6\x97\xa5\xe6\x9c\xac"));
 
-            Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse("0\n1"));
-            Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse("'"));
+            Assert.Equal("0", Parsers.UnquotedValue.Parse("0\n1").Value);   // value ends on linebreak
```

Comment:
So, are these desirable changes? I explicitly want the system to throw up on bad input. Are there other examples in these tests already that are like these cases, where bad input should be silently processed and discarded?

Reply:
I will have a closer look here.
I don't think I silently process and discard values, but the tests did not succedd in this form due to the changes.
I just forgot about these tests, I should have replaced them approprietly.

Reply:
I will just add the both removed tests again, they still work.
I think I just forgot about them, I removed them at a point where I did not fully understand how these parsers work, and added my new tests afterwards.

If you force UnquotedValue to parse to end it fails of course, because the linebreak and text in the new line are not expected.

Explanation below, I will update the test to explain the intention better.

First additional test (value ends on linebreak):
If you just try to get the valid value before the linebreak (so not End, just parse) it should of course ignore the next line.
The next line is an invalid line then of course, but that's handled by analysis of the next line, the unquoted value of our tested line ist still perfectly valid.

Second additional test (unmatched singleQuote):
With just parsing an unquotedValue should not allow a leading singleQuote, so it stops directly and returns empty string.
The unmatched doubleQuote will then fail with the next parsers, because after having a value it's not allowed to have a singlequote on the same line.
For these cases we can also argue to throw, that could remove the requirement to have quoted parsers before unquoted in ValueParser I guess.

Reply:
I double checked and this test passes already in `master` because without `End()` it just means "this is how much it consumed" and the newline is not consumed. Maybe that's a useful test to add -- I added it to my PR out of respect for your enthusiasm for it -- but it is not testing new behavior.

Reply:
Here the following two comments apply: https://github.com/tonerdo/dotnet-env/pull/87#discussion_r1437763489 and https://github.com/tonerdo/dotnet-env/pull/87#discussion_r1437790418

As I wrote there too, this test shows the "real" behavior better than the "End"-Tests, but after all I guess we do not need to be on the same page about everything ;)